modules/node/content/cms_node.e

```diff
@@ -211,6 +217,41 @@ feature -- Element change
 			auther_set: author = u
 		end
 
+	mark_not_published
+			-- Set status to not_published
+		do
+			set_status ({CMS_NODE_CONSTANTS}.not_published)
+		ensure
+			status_not_published: status = {CMS_NODE_CONSTANTS}.not_published
+		end
+
+	mark_published
+			-- Set status to published
+		do
+			set_status ({CMS_NODE_CONSTANTS}.published)
+		ensure
+			status_published: status = {CMS_NODE_CONSTANTS}.published
+		end
+
+	mark_trash
+			-- Set status to published
+		do
+			set_status ({CMS_NODE_CONSTANTS}.trash)
+		ensure
+			status_trash: status = {CMS_NODE_CONSTANTS}.trash
+		end
+
+
+feature {NONE} -- Implementation
+
+	set_status (a_status: like status)
+			-- Assign `status' with `a_status'
```

Comment:
missing, final dot.
Maybe add comment referencing the {CMS_NODE_CONSTANTS} or similar interface